### ISSUE TITLE
Exception: The specified value is not valid in the 'SslProtocolType' enumeration

### ISSUE THREAD

**Issue by timomta** [Azure/aspnet-redis-providers#189]
Hello,

I am porting a customer .Net FW 4.8 app out to Azure and it uses session state. I've been struggling to get the Redis provider to work. I'm using 5.0.0 of the session state package. It pulls in the StackExchange.Redis 2.6.96 package and I tried updating it to 2.6.111, but I get the same error.

           <Data>RedisConnectionException</Data>
            <Data>InternalFailure on {my cache server}.redis.cache.usgovcloudapi.net:6380/Interactive, Initializing/NotStarted, last: NONE, origin: ConnectedAsync, outstanding: 0, last-read: 0s ago, last-write: 0s ago, keep-alive: 60s, state: Connecting, mgr: 10 of 10 available, last-heartbeat: never, global: 0s ago, v: 2.6.111.64013

The specified value is not valid in the 'SslProtocolType' enumeration.
Parameter name: sslProtocolType
   at System.Net.Security.SslState.ValidateCreateContext(Boolean isServer, String targetHost, SslProtocols enabledSslProtocols, X509Certificate serverCertificate, X509CertificateCollection clientCertificates, Boolean remoteCertRequired, Boolean checkCertRevocationStatus, Boolean checkCertName)
   at System.Net.Security.SslStream.AuthenticateAsClient(String targetHost, X509CertificateCollection clientCertificates, SslProtocols enabledSslProtocols, Boolean checkCertificateRevocation)
   at StackExchange.Redis.ExtensionMethods.AuthenticateAsClient(SslStream ssl, String host, Nullable`1 allowedProtocols, Boolean checkCertificateRevocation)
   at StackExchange.Redis.PhysicalConnection.&lt;ConnectedAsync&gt;d__106.MoveNext()

The application is set to use Tls 1.2 in Global.asax:Application_Start with
         ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12

The Redis server is set to require Tls 1.2 Advanced Settings Minimum TLS version.

I can't seem to get past this or see what is wrong. Any ideas?

**Comment by philon-msft** [comment]
Does the Redis connection string specify an `sslProtocols` setting? Configuration options docs [here](https://stackexchange.github.io/StackExchange.Redis/Configuration.html#configuration-options), valid values [here](https://learn.microsoft.com/en-us/dotnet/api/system.security.authentication.sslprotocols?view=net-7.0#fields)

If that doesn't point to a solution, this may be a question for the https://github.com/StackExchange/StackExchange.Redis repo

**Comment by timomta** [comment]
Thanks! I really appreciate those references. 
In my case, I narrowed it down to configuration discrepancy between the runtime and compilation frameworks. I set them the same:
    <compilation debug="true" targetFramework="4.8.1" />
    <httpRuntime targetFramework="4.8.1" />

Since it was an old app, they were set to different values, which apparently wasn't friendly to the StackExchange client.
Setting them both to the same Framework version cleared up the failures.
